### PR TITLE
stylo: various fixes to improve style logging in opt builds

### PR DESCRIPTION
**Pull request by bholley**
This adds a traversal time entry to the style statistics, and switches to warn! as discussed in [1].

[1] https://bugzilla.mozilla.org/show_bug.cgi?id=1339176

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15557)

### CODE REVIEW

File: components/style/context.rs [hunk 1]
```diff
@@ -120,17 +121,22 @@ pub struct TraversalStatistics {
...
+    /// Time spent in the traversal, in milliseconds.
+    pub traversal_time_ms: f64,
 }
 
 /// Implementation of Add to aggregate statistics across different threads.
 impl<'a> Add for &'a TraversalStatistics {
     type Output = TraversalStatistics;
     fn add(self, other: Self) -> TraversalStatistics {
+        debug_assert!(self.traversal_time_ms == 0.0 && other.traversal_time_ms == 0.0,
+                      "traversal_time_ms should be set at the end by the caller");
```
Comment (emilio) [inline components/style/context.rs]: Seems to me this shouldn't live in `TraversalStatistics`, though I see why you're doing it.

**Review by emilio — APPROVED** [review]

File: components/style/context.rs (hunk 1 above)
Comment (bholley) [inline components/style/context.rs]: Yeah, I just wanted it to show up in the same perf block for @shinglyu's automation, and this seemed like the simplest way to do it.

I agree it's a bit gross. The other way I considered was that each TraversalStatistics would capture a start time when instantiated, and then we'd take the min when we merge them, and rely on the fact that we must instantiate at least one ThreadLocalStyleContext (and therefore TraversalStatistics) very near to the beginning of the traversal. But that seemed like it required a lot of knowledge about the style system for people to verify that it was measuring the right thing, so I went with the dumb approach.

**Pull request merged**